File: CODIFICACAO/BackEndMarqFacil/src/main/java/br/com/marqfacil/api/services/CategoriaService.java
package br.com.marqfacil.api.services;

import java.util.List;

import br.com.marqfacil.api.entities.Categoria;

public interface CategoriaService {

	/**
	 * Metodo para listar todas as categorias
	 * 
	 * @param idempresa
	 * @return
	 */
	List<Categoria> listAllCategorias();

	/**
	 * Categorias por idcategoria
	 * 
	 * @param idcategoria
	 * @return
	 */
	Categoria listCategoriaById(Integer idcategoria);

	/**
	 * Metodo para salvar ou alterar uma categoria
	 * 
	 * @param categoria
	 * @return
	 */
	Categoria salvarCategoria(Categoria categoria);

	/**
	 * Metodo para remover uma categoria
	 * 
	 * @param idcategoria
	 */
	void removerCategoria(Integer idcategoria);

	/**
	 * Metodo para listar categoria por tipo
	 * 
	 * @param tipo
	 * @return
	 */
	List<Categoria> listarPorTipoEIdempresa(Integer tipo, Integer idempresa);

}

File: CODIFICACAO/AppMarqFacil/src/data/services/CategoriaService.js
import { ApiService } from './ApiService'

export const CategoriaService = {
    // Rota para alterar uma categoria.
    alterarCategoria(id, data, token) {
        return ApiService.putToken(`categorias/operacao/${id}`, data, token)
    },
    // Rota para cadastrar uma nova categoria.
    salvarCategoria(data, token) {
        return ApiService.postToken(`categorias/operacao`, data, token)
    },
    // Rota para remover uma categoria.   
    removerCategoria(id, token) {
        return ApiService.deleteToken(`categorias/operacao/${id}`, token)
    },
    // Rota para listar todas as categorias.
    listarCategorias() {
        return ApiService.get(`categorias`)
    },
    // Rota para listar categorias pelo tipo.    
    listarCategoriasTipo(tipo, idempresa) {
        return ApiService.get(`categorias/${tipo}/empresa/${idempresa}`)
    },
    // Rota para listar categoria pelo idcategoria.
    listaCategoriaById(id) {
        return ApiService.get(`categorias/${id}`)
    }
}